### PR TITLE
Settings 2471693

### PR DESCRIPTION
**Pull request by Sentei71**
See https://www.drupal.org/node/2471693

### CODE REVIEW

**Comment by klausi** [comment]
Tests are failing because you forgot to use short array syntax, a coding standard requirement in Rules.

File: config/install/rules.settings.yml [hunk 1]
```diff
@@ -1,4 +1,2 @@
-log_errors: warning
-debug_log: false
-debug: false
-log_level: info
+debug_screen_log: false
+log_level: warning
```
Comment (klausi) [inline config/install/rules.settings.yml:2]: Looks like the default value for "log" is missing here?

File: config/schema/rules.schema.yml [hunk 2]
```diff
@@ -193,15 +193,9 @@ rules.settings:
...
       type: boolean
-      label: 'Log debug information to the available loggers'
-    debug:
-      type: boolean
-      label: 'Show debug information'
+      label: 'Log debug information'
     log_level:
-      type: integer
+      type: string
       label: 'Log level'
```
Comment (klausi) [inline config/schema/rules.schema.yml:201]: looks like "debug_screen_log" is missing here?

File: src/Form/ReactionRuleEditForm.php [hunk 3]
```diff
@@ -65,6 +65,12 @@ public function form(array $form, FormStateInterface $form_state) {
   protected function actions(array $form, FormStateInterface $form_state) {
     $actions = parent::actions($form, $form_state);
     $actions['submit']['#value'] = $this->t('Save');
+    $actions['cancel'] = [
+      '#type' => 'submit',
+      '#value' => $this->t('Cancel'),
+      '#submit' => ['::cancel'],
```
Comment (klausi) [inline src/Form/ReactionRuleEditForm.php:71]: Unrelated change, should be removed from this pull request.

File: src/Form/ReactionRuleEditForm.php [hunk 4]
```diff
@@ -80,6 +86,19 @@ public function save(array $form, FormStateInterface $form_state) {
...
 
   /**
+   * Form submission handler for the 'cancel' action.
+   *
+   * @param array $form
+   *   An associative array containing the structure of the form.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   The current state of the form.
+   */
+  public function cancel(array $form, FormStateInterface $form_state) {
```
Comment (klausi) [inline src/Form/ReactionRuleEditForm.php:96]: Unrelated change, should be removed from this pull request.

File: src/Form/RulesSettingsForm.php [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+      ],
+      '#default_value' => $config->get('log_level') ? $config->get('log_level') : LogLevel::WARNING,
+      '#description' => $this->t('Evaluations errors are logged to available loggers.'),
+      '#states' => [
+        // Hide the log_level radios when the debug log is disabled.
+        'invisible' => [
+          'input[name="log"]' => ['checked' => FALSE],
+        ],
+      ],
+    ];
```
Comment (klausi) [inline src/Form/RulesSettingsForm.php:59]: Looks like a checkbox to log errors to the screen is missing here?

File: config/install/rules.settings.yml (hunk 1 above)
Comment (DiDebru) [inline config/install/rules.settings.yml:2]: which default value should i set for "log"?

File: src/Form/RulesSettingsForm.php (hunk 5 above)
Comment (DiDebru) [inline src/Form/RulesSettingsForm.php:59]: Do you want a checkbox there that act similiar as the log checkobox?

if ($this->config->get('log')) {
      if ($this->levelTranslation[$this->config->get('log_level')] >= $this->levelTranslation[$level]) {
        parent::log($level, $message, $context);
      }
    }

**Comment by klausi** [comment]
continued in #372
